midenc-session/src/outputs.rs

```diff
@@ -64,21 +100,29 @@ impl OutputType {
         )
     }
 
-    pub fn all() -> [OutputType; 6] {
+    pub fn all() -> [OutputType; 7] {
         [
             OutputType::Ast,
+            OutputType::Wat,
             OutputType::Hir,
             OutputType::Masm,
             OutputType::Mast,
             OutputType::Masl,
             OutputType::Masp,
         ]
     }
+
+    /// Returns the subset of [OutputType] values considered "intermediate" for convenience
+    /// emission (WAT, HIR, MASM).
+    pub fn inter() -> [OutputType; 3] {
```

Comment:
I'd make the return type `&'static [OutputType]` here, and mark it `const`, that way we don't have to hardcode the size of the array in the signature, and it's just as useful at compile-time as an array.